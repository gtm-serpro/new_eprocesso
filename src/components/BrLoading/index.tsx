import classNames from "classnames";

export type BrLoadingProps = {
  label?: string;
  large?: boolean;
};

const BrLoading = ({ label, large: medium }: BrLoadingProps) => {
  return (
    <div className="d-flex flex-column align-items-center">
      <div className={classNames("loading", medium && "medium")}></div>
      {label && <span className="rotulo mt-1">{label}</span>}
    </div>
  );
};

export default BrLoading;
