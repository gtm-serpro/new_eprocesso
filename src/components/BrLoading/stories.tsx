import { Meta, Story } from "@storybook/react";

import BrLoading, { BrLoadingProps } from ".";

export default {
  title: "Feedback/Loading",
  component: BrLoading,
} as Meta;

const Template: Story<BrLoadingProps> = (args) => <BrLoading {...args} />;

export const Padrao = Template.bind({});
Padrao.storyName = "Padrão";

export const Large = Template.bind({});
Large.args = { large: true };

export const ComRotulo = Template.bind({});
ComRotulo.storyName = "Com Rótulo";
ComRotulo.args = { label: "Carregando..." };
