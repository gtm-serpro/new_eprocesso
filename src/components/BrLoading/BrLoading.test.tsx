import { render, screen } from "@testing-library/react";

import BrLoading from ".";

describe("<BrLoading />", () => {
  it("deveria renderizar o loading simples", () => {
    const { container } = render(<BrLoading />);
    expect(container.getElementsByClassName("loading")).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar o loading grande se prop informada", () => {
    const { container } = render(<BrLoading large />);
    const element = container.getElementsByClassName("loading");
    expect(element[0]).toHaveClass("medium");
  });

  it("nao deveria renderizar rótulo se nao informado", () => {
    render(<BrLoading />);
    const element = screen.queryByText(/rótulo carregando.../i);
    expect(element).toBeNull();
  });

  it("deveria renderizar rótulo se informado", () => {
    render(<BrLoading label="Rótulo Carregando..." />);
    const element = screen.getByText(/rótulo carregando.../i);
    expect(element).toBeInTheDocument();
    expect(element).toHaveClass("rotulo");
  });
});
