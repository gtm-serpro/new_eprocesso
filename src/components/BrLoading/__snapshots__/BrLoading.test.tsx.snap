// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrLoading /> deveria renderizar o loading simples 1`] = `
<div
  class="d-flex flex-column align-items-center"
>
  <div
    class="loading"
  />
</div>
`;
